### PR TITLE
satellite/gracefulexit: increment failed count for "successful" transfers that failed satellite validation

### PR DESCRIPTION
**Pull request by mobyvb**
What: 
When a node gracefully exiting tells the satellite that it successfully transferred a piece, but the transfer fails satellite validation due to incorrect hash signatures, mismatching hashes, etc... We should mark those transfer queue items as having a failure.

Why:
Our failure counts for graceful exit should be accurate.

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

**Comment by egonelbre** [comment]
AFAIR, when there's an error with hash validation or any of the content then the exiting node should be disqualified because it provided wrong information during graceful exit.

File: satellite/gracefulexit/endpoint.go [hunk 1]
```diff
@@ -497,6 +497,16 @@ func (endpoint *Endpoint) handleSucceeded(ctx context.Context, pending *pendingM
 		return Error.New("Transfer path cannot be nil")
 	}
 
+	defer func() {
```
Comment (egonelbre) [inline satellite/gracefulexit/endpoint.go]: Try to avoid defers for logic control since they make the code much harder to read.

**Comment by mobyvb** [comment]
> AFAIR, when there's an error with hash validation or any of the content then the exiting node should be disqualified because it provided wrong information during graceful exit.

I think you are right. But the exiting storagenode is not currently validating the recipient storagenode's hash and signature, which means it does not necessarily know the data it's sending the satellite is invalid. I need to figure out how to get the receiving node's peer ID on the exiting node, then the validation should be easy and we can go ahead and DQ nodes that send bad data.

**Comment by egonelbre** [comment]
The exiting storagenode must validate it.

Exiting storagenode makes a connection to the replacement storagenode, so it has all the necessary information to verify things.

**Comment by mobyvb** [comment]
@egonelbre closing this PR. The changes we discussed are in https://github.com/storj/storj/pull/3371

**Pull request closed without merging**